crates/forge-runner/src/lib.rs

```diff
@@ -0,0 +1,442 @@
+pub struct RunnerConfig {
+    pub workspace_root: Utf8PathBuf,
+    pub exit_first: bool,
+    pub fork_targets: Vec<ForkTarget>,
+    pub fuzzer_runs: u32,
+    pub fuzzer_seed: u64,
+}
+
+impl RunnerConfig {
+    /// Creates a new `RunnerConfig` from given arguments
+    ///
+    /// # Arguments
+    ///
+    /// * `test_name_filter` - Used to filter test cases by names
+    /// * `exact_match` - Should test names match the `test_name_filter` exactly
+    /// * `exit_first` - Should runner exit after first failed test
+    #[must_use]
+    #[allow(clippy::too_many_arguments, clippy::fn_params_excessive_bools)]
+    pub fn new(
+        workspace_root: Utf8PathBuf,
+        exit_first: bool,
+        fork_targets: Vec<ForkTarget>,
+        fuzzer_runs: u32,
+        fuzzer_seed: u64,
+    ) -> Self {
+        Self {
+            workspace_root,
+            exit_first,
+            fork_targets,
+            fuzzer_runs,
+            fuzzer_seed,
+        }
+    }
+}
+
+pub struct RunnerParams {
+    corelib_path: Utf8PathBuf,
+    contracts: HashMap<String, StarknetContractArtifacts>,
+    predeployed_contracts: Utf8PathBuf,
+    environment_variables: HashMap<String, String>,
+    linked_libraries: Vec<LinkedLibrary>,
+}
+
+impl RunnerParams {
+    #[must_use]
+    pub fn linked_libraries(&self) -> &Vec<LinkedLibrary> {
+        &self.linked_libraries
+    }
+    #[must_use]
+    pub fn corelib_path(&self) -> &Utf8PathBuf {
+        &self.corelib_path
+    }
+}
+
+impl RunnerParams {
+    #[must_use]
+    pub fn new(
+        corelib_path: Utf8PathBuf,
+        contracts: HashMap<String, StarknetContractArtifacts>,
+        predeployed_contracts: Utf8PathBuf,
+        environment_variables: HashMap<String, String>,
+        linked_libraries: Vec<LinkedLibrary>,
+    ) -> Self {
+        Self {
+            corelib_path,
+            contracts,
+            predeployed_contracts,
+            environment_variables,
+            linked_libraries,
+        }
+    }
+}
+
+/// Exit status of the runner
+#[derive(Debug, PartialEq, Clone)]
+pub enum RunnerStatus {
+    /// Runner exited without problems
+    Default,
+    /// Some test failed
+    TestFailed,
+    /// Runner did not run, e.g. when test cases got skipped
+    DidNotRun,
+}
+
+#[derive(Debug, Clone)]
+pub struct ValidatedForkConfig {
+    url: Url,
+    block_id: BlockId,
+}
+
+impl ValidatedForkConfig {
+    #[must_use]
+    pub fn new(url: Url, block_id: BlockId) -> Self {
+        Self { url, block_id }
+    }
+}
+
+impl TryFrom<RawForkParams> for ValidatedForkConfig {
+    type Error = anyhow::Error;
+
+    fn try_from(value: RawForkParams) -> std::result::Result<Self, Self::Error> {
+        Ok(ValidatedForkConfig {
+            url: value.url.parse()?,
+            block_id: value.block_id,
+        })
+    }
+}
+
+impl ForkConfig for ValidatedForkConfig {}
+
+pub type TestCaseRunnable = TestCase<ValidatedForkConfig>;
+
+#[derive(Debug, Clone)]
+pub struct TestCrate {
```

Comment:
Can we keep the previous name? It is not actually a `TestCrate` itself, it has test configs too

Reply:
It doesn't make sense to name it `CompiledTestCrateRunnable` if it's the only variant of this struct present in forge-runner. It's no longer related to struct from the test collector except for the similar contents.

Reply:
I disagree, when I browse the code following the reference (cmd + leftclick) I won't check every time in what package I am, it introduces additional complexity

Reply:
We can use `forge_runner::TestCrate` syntax instead of importing it directly if that's more readable. I don't think we names in other crates should heavily influence the names in this one, that's kind of agains the whole point of "namespacing" our code with different crates.

Reply:
I understand your point, but I think it just adds cognitive complexity for no reason. Those crates are really tightly coupled so keeping names unique across both of them makes a lot of sense imo

Reply:
Agree with @piotmag769 here, those are important information

Reply:
These arguments are valid in general case, though I think that we'd never name this struct `CompiledTestCrateRunnable` or whatever if we had two separate crates from the very beginning. 

This name was introduced just because we had multiple versions of a similar struct and needed to differentiate between them which is no longer the case. 

If you look at the usages of it in the code, it's pretty much constructed in one place in forge and all other usages of it are in the forge-runner, where the name doesn't matter that much.